community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/resume.py

```diff
@@ -99,19 +99,29 @@ def instance_properties(nodeset, model, placement_group, labels=None):
         props.scheduling.maintenanceInterval = nodeset.maintenance_interval
 
     if nodeset.dws_flex.enabled:
-        update_props_dws(props,nodeset.dws_flex)
+        update_props_dws(props, nodeset.dws_flex, job_id)
 
     # Override with properties explicit specified in the nodeset
     props.update(nodeset.get("instance_properties") or {})
     
     return props
 
-def update_props_dws(props:dict,dws_flex:dict) -> None:
+def update_props_dws(props:dict, dws_flex:dict, job_id=None) -> None:
     props.scheduling.onHostMaintenance = "TERMINATE"
     props.scheduling.instanceTerminationAction = "DELETE"
-    props.scheduling.maxRunDuration['seconds'] = dws_flex.max_run_duration
     props.reservationAffinity['consumeReservationType'] = "NO_RESERVATION"
 
+    if dws_flex.use_job_duration and job_id:
+        current_job = lookup().job(job_id)
+
+        if current_job.duration<=1209600 and current_job.duration>=30:
+            props.scheduling.maxRunDuration['seconds'] = current_job.duration
```

Comment:
I think having `job.duration` should lead to `max_duration` to be short ( secs)

Reply:
Might need some additional clarification on this, should we be overriding whatever the user put as `max_run_duration` or should we restrict the ability to even set `max_run_duration` if `use_job_duration` is enabled. I would probably lean towards the latter, as overriding user input otherwise may be unexpected (from their POV).